PYIC-8778: Remove unused published-did-documents bucket and switch all envs to SPOT DID source

AmritSidhu opened the pull request:
## Proposed changes
### What changed

- Deleted SpotKeysS3Bucket and SpotKeysBucketPolicy resources referencing published-did-documents-* 
- Updated RestDidDocumentResourceS3Role to always reference govuk-one-login-spot-published-keys-${Environment}/did.json 
- Removed conditional logic and references to the deprecated DID bucket

### Why did it change

- The following bucket has been flagged for infringing ‘s3-bucket-ssl-requests-only': published-did-documents-production, because the bucket does not have an SSL deny policy in place.

### Issue tracking
<!-- Jira ticket & other docs, like RFCs -->

- [PYIC-8777](https://govukverify.atlassian.net/browse/PYIC-8777)

## Checklists

- [ ] READMEs and documentation up-to-date
- [ ] Browser/ unit/ Selenium tests have been written/ updated
- [ ] No risk of exposure: PII, credentials, etc through logs/ code
- [ ] Ensure added/updated routes have CSRF protection if required


[PYIC-8777]: https://govukverify.atlassian.net/browse/PYIC-8777?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

AmritSidhu committed:
PYIC-8778: Remove unused published-did-documents bucket and switch all envs to SPOT DID source

Deleted SpotKeysS3Bucket and SpotKeysBucketPolicy resources referencing published-did-documents-*
Updated RestDidDocumentResourceS3Role to always reference govuk-one-login-spot-published-keys-${Environment}/did.json
Removed conditional logic and references to the deprecated DID bucket

blakeyp commented on deploy/template.yaml:
```diff
@@ -1135,12 +1089,7 @@ Resources:
-                      - IsSubscriptionEnviroment
-                      - !Sub "govuk-one-login-spot-published-keys-${Environment}"
-                      - !Sub "published-did-documents-${Environment}"
+                Resource: !Sub "arn:aws:s3:::govuk-one-login-spot-published-keys-${Environment}/did.json"
```
In the lower envs I suspect this might break as I'm not sure SPOT have this configured in dev/build. I don't _think_ we're using or have a need for the DID endpoint in our dev/build envs - if this is the case we probably want to just put a Condition on this resource itself so it's only created in staging and above (I think we also need to update RestDidDocumentMethod?)

AmritSidhu committed:
PYIC-8777: Remove legacy published-did-documents resources and use SPOT DID source for subscription environments

Updated API Gateway and IAM role to use SPOT’s govuk-one-login-spot-published-keys-* bucket for staging, integration, and production only
Fix mis-spelling

In staging/integration/production → it depends on both methods
In dev/build → it skips the DID method, because it doesn’t exist

AmritSidhu committed:
PYIC-8777: Fix API Gateway deployment duplication by splitting subscription and non-subscription deployments

Separated the REST API Gateway deployment into two resources — one for subscription environments (staging, integration, production) and one for non-subscription environments (dev, build)
Updated DevApiMapping and ApiGatewayStage to depend on the correct deployment using conditional logic to prevent duplicate or conflicting deployments

blakeyp commented on deploy/template.yaml:
```diff
@@ -51,10 +51,11 @@ Conditions:
   IsNotDevelopment: !Not [ !Condition IsDevelopment ]
   IsProduction: !Equals [ !Ref Environment, production ]
-  IsSubscriptionEnviroment: !Or
+  IsSubscriptionEnvironment: !Or
```
I think this was named because staging and above are the Splunk subscription envs, but given this is used in multiple use cases it would be clearer if it was just named what it actually is i.e. `IsStagingIntProd` ?

blakeyp commented on deploy/template.yaml:
```diff
@@ -51,10 +51,11 @@ Conditions:
     - !Equals [ !Ref Environment, staging ]
     - !Equals [ !Ref Environment, integration ]
     - !Equals [ !Ref Environment, production ]
+  IsNotSubscriptionEnvironment: !Not [ !Condition IsSubscriptionEnvironment ]
```
Prefer this to be explicitly *is dev/build* ?

blakeyp commented on deploy/template.yaml:
```diff
@@ -292,13 +293,18 @@ Resources:
+      cfn-lint:
+        config:
+          ignore_checks:
+            - E3005
+            - W1001
```
Can we add a comment explaining why we're ignoring

blakeyp commented on deploy/template.yaml:
```diff
@@ -1213,13 +1169,26 @@ Resources:
 
-  RestApiGwDeployment202502241200:
+  # For environments that have the DID document (staging, integration, production)
+  RestApiGwDeployment202502241200Sub:
```
Maybe rename this `WithDidEndpoint` ? and we should probably update the `YYYYMMDDHHMM` timestamp

(plus same for "without")

blakeyp commented on deploy/template.yaml:
```diff
@@ -1213,13 +1169,26 @@ Resources:
           - REGIONAL
 
-  RestApiGwDeployment202502241200:
+  # For environments that have the DID document (staging, integration, production)
```
to clarify a little - "that host SPOT's DID endpoint" ?

blakeyp commented on deploy/template.yaml:
```diff
@@ -1213,13 +1169,26 @@ Resources:
+    Type: AWS::ApiGateway::Deployment
+    Properties:
+      RestApiId: !Ref RestApiGateway
+      Description: "Deployment without DID document for non-subscription environments"
```
Again prefer to refer to actual envs rather than "subscription"/"non-subscription" which is not well known

AmritSidhu committed:
PYIC-8777: Clarify and rename API Gateway deployment logic for DID endpoint handling

- Renamed RestApiGwDeployment resources to WithDidEndpoint / WithoutDidEndpoint with updated 202511111337 timestamp
- Updated comments and descriptions to explicitly mention SPOT’s DID endpoint
- Replaced vague “subscription/non-subscription” references with explicit environment names (staging, integration, production, dev, build)
- Added one-line explanations for cfn-lint ignores (E3005, W1001) in DevApiMapping
- Improved condition naming clarity (IsStagingIntProd / IsDevOrBuild)

AmritSidhu committed:
PYIC-8777: Add cfn-lint ignore for conditional stage deployment references

Added W1001 lint suppression to ApiGatewayStage to handle false-positive warnings caused by conditional !If references to environment-specific API Gateway deployments
Clarifies that both deployment resources (WithDidEndpoint and WithoutDidEndpoint) are mutually exclusive and safely handled by CloudFormation
